### PR TITLE
[Payum] Scalar types in Payum

### PR DESCRIPTION
**Pull request by Zales0123**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | no |
| BC breaks?      | yes |
| Related tickets | https://github.com/Sylius/Sylius/issues/8425 |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Bundle/PayumBundle/Form/Extension/CryptedGatewayConfigTypeExtension.php [hunk 1]
```diff
@@ -34,15 +34,15 @@
     /**
      * @param CypherInterface|null $cypher
      */
-    public function __construct(CypherInterface $cypher = null)
+    public function __construct(?CypherInterface $cypher = null)
```
Comment (pamil) [inline src/Sylius/Bundle/PayumBundle/Form/Extension/CryptedGatewayConfigTypeExtension.php]: Side note: we can make it not accept nulls at all and implement noop cypher as a replacement for null.

File: src/Sylius/Bundle/PayumBundle/Model/PaymentSecurityToken.php [hunk 2]
```diff
@@ -81,63 +79,63 @@ public function getDetails()
...
+    public function getHash(): string
     {
         return $this->hash;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function setHash($hash)
+    public function setHash($hash): void
```
Comment (pamil) [inline src/Sylius/Bundle/PayumBundle/Model/PaymentSecurityToken.php]: `?string $hash`

File: src/Sylius/Bundle/PayumBundle/Model/PaymentSecurityToken.php [hunk 3]
```diff
@@ -81,63 +79,63 @@ public function getDetails()
...
+    public function getTargetUrl(): string
     {
         return $this->targetUrl;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function setTargetUrl($targetUrl)
+    public function setTargetUrl($targetUrl): void
```
Comment (pamil) [inline src/Sylius/Bundle/PayumBundle/Model/PaymentSecurityToken.php]: `?string $targetUrl`

File: src/Sylius/Bundle/PayumBundle/Model/PaymentSecurityToken.php [hunk 4]
```diff
@@ -81,63 +79,63 @@ public function getDetails()
...
+    public function getAfterUrl(): string
     {
         return $this->afterUrl;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function setAfterUrl($afterUrl)
+    public function setAfterUrl($afterUrl): void
```
Comment (pamil) [inline src/Sylius/Bundle/PayumBundle/Model/PaymentSecurityToken.php]: `?string $afterUrl`

File: src/Sylius/Bundle/PayumBundle/Model/PaymentSecurityToken.php [hunk 5]
```diff
@@ -55,15 +53,15 @@ public function __construct()
...
+    public function getId(): string
     {
         return $this->hash;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function setDetails($details)
+    public function setDetails($details): void
```
Comment (pamil) [inline src/Sylius/Bundle/PayumBundle/Model/PaymentSecurityToken.php]: Missing argument typehints in this class.

Comment (Zales0123) [inline src/Sylius/Bundle/PayumBundle/Model/PaymentSecurityToken.php]: It's sadly `object` in interface :/

File: src/Sylius/Bundle/PayumBundle/Model/PaymentSecurityToken.php (hunk 4 above)
Comment (Zales0123) [inline src/Sylius/Bundle/PayumBundle/Model/PaymentSecurityToken.php]: It's sadly (this and all above) from `TokenInterface` from `Payum` which does not have any strict types.

**Comment by pamil** [comment]
Amazing work, thx m8!

**Pull request merged**